shell/browser/api/electron_api_web_contents.cc

```diff
@@ -2489,7 +2489,13 @@ double WebContents::GetZoomLevel() const {
   return zoom_controller_->GetZoomLevel();
 }
 
-void WebContents::SetZoomFactor(double factor) {
+void WebContents::SetZoomFactor(gin_helper::ErrorThrower thrower,
+                                double factor) {
+  if (factor <= 0.0) {
```

Comment:
There are all kinds of ways for comparing equality in floating-points to fail.

If we're trying to ensure that `factor` is positive, it's safer to add `#inclue <cmath>` to the top and use `std::signbit()`:

```suggestion
  if (!std::signbit(factor)) {
```

Reply:
The analysis is right, but the suggestion is wrong: `signbit()` is not going to do what we want here.

Revised suggestion: either use `std::numeric_limits<double>::epsilon()` or KISS with `< 0.001` or the like :slightly_smiling_face: